index.html

```diff
@@ -2117,7 +2117,7 @@ <h2>Forms</h2>
                         <br><br>
 
                         <div class="custom-file">
-                            <input type="file" class="custom-file-input" id="file3">
+                            <input type="file" class="custom-file-input" id="file3" multiple data-multiple-files-text="files">
```

Comment:
> `data-multiple-files-text="files"`

This is going to be a royal pain in the ass to localize.

First, we need to pass a transchoice compatible string here (see [Pluralization](https://symfony.com/doc/3.4/translation.html#pluralization))

```
{0} no files|{1} 1 file|]1,Inf[ %count% files
```

Then, it should be parsed according to the number of elements using something like this:
https://github.com/rmariuzzo/Lang.js/blob/master/src/lang.js#L178-L228

Reply:
Hi @eternoendless  - just want to confirm the task and check if I understand everything correctly.

1. I'll pass a translation withing the data attribute like:
```twig
<input data-pluralised-text="{{ '{0} no files|{1} 1 file|]1,Inf[ %count% files'|trans({}, 'Admin.Global') }}">
```

2. I also need a locale so I'll pass it with another data attribute
3. I'll use the code from the link you have provided - like 90% of the stuff is similar

Reply:
> 2. I also need a locale so I'll pass it with another data attribute

No, you don't need a locale because by using "trans" in Twig the whole string is translated without making a choice on what pattern to use.

For example

```twig
{{ '{0} no files|{1} 1 file|]1,Inf[ %count% files'|trans({}, 'Admin.Global') }}"
```

Should output in french:

```
{0} aucun fichier|{1} 1 fichier|]1,Inf[ %count% fichiers
```

Then we'll use that with transchoice to figure out which of the three choices we'll use, according to the number of `files`.

eg. with "1" file:

```
1 fichier
```

UI kit of course knows nothing of translation, it should only be able to make a choice out of the provided, **already translated** pattern.

Reply:
@eternoendless  I have analysed a bit this part https://github.com/rmariuzzo/Lang.js/blob/master/src/lang.js#L494 and tried to imagine how it would look like in my own language to translate it:

there are two plural forms in Lithuanian (LT) and one singular form:

for numers like 21,31,41,51,61 etc... it uses singular form **failas**
for numbers like 22,23,24,25... it uses **failai**
for numbers like 30,40,50,60,70... it uses **failų**

its impossible to define all rules  so here its the certain math applied , but only for my language case. The code in https://github.com/rmariuzzo/Lang.js/blob/master/src/lang.js#L494 solves it but it requires to know the iso code.

My question is - are we really don't need to know the locale because without using the locale we won't get the correct results in all possible cases. :thinking:

Reply:
I guess we'll have to support the locale code too. Just make it optional and fall back to 'en' if not provided.